### DISCUSSION TITLE
PaddleOCR/community/community_contribution

### DISCUSSION THREAD

**Discussion by giscus** [PaddlePaddle/PaddleOCR#13657] [Q&A]
# PaddleOCR/community/community_contribution

Awesome multilingual OCR toolkits based on PaddlePaddle (practical ultra lightweight OCR system, support 80+ languages recognition, provide data annotation and synthesis tools, support training and deployment among server, mobile, embedded and IoT devices)

https://paddlepaddle.github.io/PaddleOCR/community/community_contribution.html

**Comment by WaterMirror1** [comment]
您好，我这里使用paddleOCR的C++版本进行本地英文和数字识别任务的部署成功后，想要使用tensorrt来进行推理加速，我将推理模型已经转换成了.onnx，但是在ocr_cls.cpp、ocr_det.cpp、ocr_rec.cpp文件中的模型读取接口，发现只可以读取.pdmodel格式的模型，好像不可以读取.onnx的模型，这种情况应该怎么处理比较好呢？这里是正常的用法：  paddle_infer::Config config;
  config.SetModel(model_dir + "/inference.pdmodel",
                  model_dir + "/inference.pdiparams");
但是现在我想要使用tensorrt来进行加速，这种方式去加载我的tensorrt格式的模型就有问题了：
    config.SetModel(model_dir + "/onnxmodel.onnx");
有如下报错：NotFoundError: Cannot open file model/whl/det/en/en_PP-OCRv3_det_infer/onnxmodel.onnx/__model__, please confirm whether the file is normal.
  [Hint: Expected static_cast<bool>(fin.is_open()) == true, but received static_cast<bool>(fin.is_open()):0 != true:1.] (at ..\paddle\fluid\inference\api\analysis_predictor.cc:2578)

**Reply by GreatV** [reply]
观察一下报错信息，模型目录是不是不对。

**Reply by WaterMirror1** [reply]
模型目录我是和原来的.pdmodel模型在同一目录下的，模型目录应该是没有问题的。你的意思是config.SetModel这种方式也是可以直接读取.onnx格式的模型对吗？

**Reply by GreatV** [reply]
因为报错是

> Cannot open file `model/whl/det/en/en_PP-OCRv3_det_infer/onnxmodel.onnx/model`

**Reply by WaterMirror1** [reply]
嗯，我明白的，这个存放模型的目录是在Release/build目录下放的model，在args.cpp中我给三个模型路径写的都是model/...这样的路径，已经验证过正常寻找.pdmodel模型时是可以找到的。只是当我改成使用.onnx格式的模型时，在onnxmodel.onnx的后面还多了一个/__model__，这个东西我不清楚是哪里来的。Cannot open file model/whl/det/en/en_PP-OCRv3_det_infer\onnxmodel.onnx/__model__

**Reply by WaterMirror1** [reply]
额，路径的'/'没标识好，不好意思可能造成您的误解，实际上路径是这样的：Cannot open file model/whl/det/en/en_PP-OCRv3_det_infer/onnxmodel.onnx/__model__，但是仍然是一样的报错

**Reply by GreatV** [reply]
可以试试只传目录

**Reply by WaterMirror1** [reply]
您好，刚刚我试过了这个方法，不过还是一样的问题：NotFoundError: Cannot open file model/whl/det/en/en_PP-OCRv3_det_infer/__model__, please confirm whether the file is normal.

**Reply by GreatV** [reply]
如果是用cpp_infer改的，目录应该不用改吧，`use_tensorrt=true`应该就好了。

**Reply by WaterMirror1** [reply]
我使用的cpp_infer改的，使用的官网提供的英文推理模型。您的意思是我想要用tensorrt加速的话，并不需要将pytorch格式的模型转成.onnx格式的模型吗？我在args.cpp中已经改成true了的，DEFINE_bool(use_tensorrt, true, "Whether use tensorrt.");
其他的和目录有关的部分，我后来转换的.onnx模型和原来下载下来的.pdmodel模型是没做任何修改的。

**Reply by GreatV** [reply]
是的，paddle inference应该会自动处理这部分。
